### PR TITLE
Add change log for 0.5.1 release

### PR DESCRIPTION
**Pull request by Jiawei0227**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add change log for 0.5.1 release

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/cc @msau42

### CODE REVIEW

File: CHANGELOG-0.5.md [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+
+[Documentation](https://docs.k8s.io/docs/home)
+# Changelog since v0.5.0
+
+## Changes by Kind
+
+### Feature
+
+- Add `process_start_time_seconds` metric. ([#106](https://github.com/kubernetes-csi/external-resizer/pull/106), [@Jiawei0227](https://github.com/Jiawei0227))
+- Adding configurable QPS and Burst to k8s client and use a separate client for leader election. ([#91](https://github.com/kubernetes-csi/external-resizer/pull/91), [@RaunakShah](https://github.com/RaunakShah))
```
Comment (msau42) [inline CHANGELOG-0.5.md:11]: hmm something doesn't seem right with the release notes tool. these two changes were only in 1.0?

File: CHANGELOG-0.5.md [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+- Pods watch is an expensive operation. Hence make the feature of handling volume in use errors optional, by registering pod informer based on a feature gate ([#89](https://github.com/kubernetes-csi/external-resizer/pull/89), [@saikat-royc](https://github.com/saikat-royc))
+
+### Uncategorized
+
+- Build with Go 1.15 ([#98](https://github.com/kubernetes-csi/external-resizer/pull/98), [@pohly](https://github.com/pohly))
+- Publishing of images on k8s.gcr.io ([#84](https://github.com/kubernetes-csi/external-resizer/pull/84), [@pohly](https://github.com/pohly))
```
Comment (msau42) [inline CHANGELOG-0.5.md:17]: Oh hm we need to backport some more changes to actually make this work.

Comment (msau42) [reply CHANGELOG-0.5.md:17]: I think it's going to be too much effort to backport the cloudbuild changes and fix the current multiarch build failure. I propose that we release these patches using the old travis method, and then copy the images over to gcr

File: CHANGELOG-0.5.md (hunk 1 above)
Comment (Jiawei0227) [reply CHANGELOG-0.5.md]: The issue is related to https://github.com/kubernetes-csi/csi-release-tools/issues/7#issuecomment-705128427

Fixed this by manually updated the release notes.

File: CHANGELOG-0.5.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+[Documentation](https://docs.k8s.io/docs/home)
+# Changelog since v0.5.0
+
+## Changes by Kind
+
+### Feature
+
+### Uncategorized
+
+- release-0.5: update release-tools ([#110](https://github.com/kubernetes-csi/external-resizer/pull/75), [@Jiawei0227](https://github.com/Jiawei0227))
```
Comment (msau42) [inline CHANGELOG-0.5.md:12]: Can you add a note here saying this also updates the go version to 1.15?

**Pull request merged**